pkg/platform/kube/dlx/handler.go

```diff
@@ -25,17 +25,46 @@ func NewHandler(logger logger.Logger, functionStarter *FunctionStarter) (Handler
 }
 
 func (h *Handler) handleRequest(res http.ResponseWriter, req *http.Request) {
+	var targetURL *url.URL
+	var err error
+	var functionName string
+
 	responseChannel := make(chan FunctionStatusResult, 1)
 	defer close(responseChannel)
 
-	headerTarget := req.Header.Get("X-nuclio-target")
-	if headerTarget == "" {
-		h.logger.Warn("Must pass X-nuclio-target header value")
-		res.WriteHeader(http.StatusBadRequest)
-		return
+	// first try to see if our request came from ingress controller
+	forwardedHost := req.Header.Get("X-Forwarded-Host")
+	forwardedPort := req.Header.Get("X-Forwarded-Port")
+	originalURI := req.Header.Get("X-Original-Uri")
+	functionName = req.Header.Get("X-Service-Name")
+
+	if forwardedHost != "" && forwardedPort != "" && functionName != "" {
```

Comment:
log + bad request error duplicated. 
move to a function that only returns nuclio ErrorWithStatus, return status code if error + log.